### PR TITLE
[1.4] Re-added extensive DustID.cs and added DustID.TML.cs

### PR DESCRIPTION
**Pull request by RighteousRyan1**
I re-addition of DustID extension by @SmugBlanco. Might need changes in the future, but for now, this should be good.

### CODE REVIEW

**Review by Mirsario — CHANGES_REQUESTED** [review]

File: patches/tModLoader/Terraria/IngameOptions.cs.patch [hunk 1]
```diff
@@ -77,6 +77,15 @@
  			if (DrawLeftSide(sb, Lang.menu[118].Value, num10, vector, vector2, leftScale)) {
  				leftHover = num10;
  				if (flag4)
+@@ -276,6 +_,8 @@
+ 			if (DrawLeftSide(sb, Lang.inter[35].Value, num10, vector, vector2, leftScale)) {
+ 				leftHover = num10;
+ 				if (flag4) {
++					SystemLoader.PreSaveAndQuit();
++					// Calls before anything else is
+ 					Close();
```
[the remark this answers is not in the document]
Comment (RighteousRyan1) [reply patches/tModLoader/Terraria/IngameOptions.cs.patch:86]: Don't know why that's there at all

[the remark this answers is not in the document]
Comment (RighteousRyan1) [reply patches/tModLoader/Terraria/IngameOptions.cs.patch:86]: i already implemented SystemLoader.PreSaveAndQuit in another PR

**Comment by RighteousRyan1** [comment]
Closed since I was mysteriously a months behind in pull requests.

**Comment by JavidPack** [comment]
Isn't this fine to Squash and Merge now?

**Pull request closed without merging**